src/python/pants/rules/core/test_cloc.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Sequence
+
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.rules.core import cloc
+from pants_test.console_rule_test_base import ConsoleRuleTestBase
+
+
+class ClocTest(ConsoleRuleTestBase):
+  goal_cls = cloc.CountLinesOfCode
+
+  @classmethod
+  def rules(cls):
+    return super().rules() + cloc.rules()
+
+  @classmethod
+  def alias_groups(cls):
+    return BuildFileAliases(targets={'java_library': JavaLibrary})
+
+  def assert_counts(self, result: Sequence[str], lang: str, num_files: int, blank: int, comment: int, code: int) -> None:
```

Comment:
Only an FYI, not gonna make you do this. You've pushed enough changes. One of my favorite features in Python is requiring kwarg only by specifying `*`:

```suggestion
  def assert_counts(self, result: Sequence[str], lang: str, *, num_files: int, blank: int, comment: int, code: int) -> None:
```

This means that anything after the star must use a keyword argument. You end up using kwargs anyways, but this would mean that the language itself enforces it for you.

Reply:
That's a neat feature I didn't know about and I'll keep it in mind, but since this is test support code that we don't expect to use outside this test I think it's okay to leave this out.